src/Laravel/Eloquent/State/LinksHandler.php

```diff
@@ -37,22 +40,72 @@ public function handleLinks(Builder $builder, array $uriVariables, array $contex
 
                     continue;
                 }
 
                 if ($from = $link->getFromProperty()) {
                     $relation = $this->application->make($link->getFromClass());
-                    $builder = $builder->getModel()->where($builder->getModel()->getTable().'.'.$relation->{$from}()->getForeignKeyName(), $identifier);
+                    $builder = $builder->getModel()->where($relation->{$from}()->getQualifiedForeignKeyName(), $identifier);
 
                     continue;
                 }
 
-                $builder->where($builder->getModel()->getTable().'.'.$link->getIdentifiers()[0], $identifier);
+                $builder->where($builder->getModel()->qualifyColumn($link->getIdentifiers()[0]), $identifier);
             }
 
             return $builder;
+        } elseif ($operation instanceof GraphQlOperation) {
+            if (!($linkClass = $context['linkClass'] ?? false)) {
+                return $builder;
+            }
+
+            $newLink = null;
+            $linkProperty = $context['linkProperty'] ?? null;
+
+            try {
+                $resourceMetadataCollection =  $this->application->make(ResourceMetadataCollectionFactoryInterface::class)->create($linkClass);
+                $linkedOperation = $resourceMetadataCollection->getOperation($operation->getName());
+            } catch (OperationNotFoundException $e) {
+                // Instead, we'll look for the first Query available.
+                foreach ($resourceMetadataCollection as $resourceMetadata) {
+                    foreach ($resourceMetadata->getGraphQlOperations() as $op) {
+                        if ($op instanceof Query) {
+                            $linkedOperation = $op;
+                        }
+                    }
+                }
+            }
+
+            if (!$linkedOperation) {
+                return $builder;
+            }
+
+            $resourceClass = $builder->getModel()::class;
+            foreach ($linkedOperation->getLinks() ?? [] as $link) {
+                if ($resourceClass === $link->getToClass() && $linkProperty === $link->getFromProperty()) {
+                    $newLink = $link;
+                    break;
+                }
+            }
+
+            if (!$newLink) {
+                return $builder;
+            }
+
+            $identifier = $uriVariables[$newLink->getIdentifiers()[0]];
+
+            if ($to = $newLink->getToProperty()) {
+                return $builder->where($builder->getModel()->{$to}()->getQualifiedForeignKeyName(), $identifier);
+            }
+
+            if ($from = $newLink->getFromProperty()) {
+                $relation = $this->application->make($newLink->getFromClass());
+                return $builder->getModel()->where($relation->{$from}()->getQualifiedForeignKeyName(), $identifier);
+            }
+
+            $builder->where($builder->getModel()->qualifyColumn($newLink->getIdentifiers()[0]), $identifier);
```

Comment:
this looks quite nice, and quite similar to the code we use on HTTP operations, maybe that we can use a function with the code that's similar, the lines I'm refering to are:

```php
if ($to = $newLink->getToProperty()) {
    return $builder->where($builder->getModel()->{$to}()->getQualifiedForeignKeyName(), $identifier);
}

if ($from = $newLink->getFromProperty()) {
    $relation = $this->application->make($newLink->getFromClass());
    return $builder->getModel()->where($relation->{$from}()->getQualifiedForeignKeyName(), $identifier);
}

$builder->where($builder->getModel()->qualifyColumn($newLink->getIdentifiers()[0]), $identifier);
```

(unless I'm mistaken which is also possible :sweat_smile:)

Reply:
Nah it actually is the same, it is what i meant by "cleanup" i will do it today, so this PR is ready soon